### ISSUE TITLE
32-bit GpuBufferDeviceAddressOOB hangs with VK_ERROR_DEVICE_LOST if run in parallel with 64-bit tests on NVIDIA 527.77

### ISSUE THREAD

**Issue by lunarpapillo** [KhronosGroup/Vulkan-ValidationLayers#4967]
## Environment

This was observed after installing the NVIDIA 527.77 driver on a **Windows** system with an **NVIDIA RTX 2080 Ti** GPU .  It's not clear to me whether or how much the driver and/or GPU are involved, though the problem has not been observed on a machine with the 472.92 driver.  

## Symptoms

Running a 32-bit `vk_layer_validation_tests.exe` in isolation runs perfectly fine, as does running a 64-bit test in isolation.

If you run both at the same time, the 32-bit test will hang when the `VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB` test receives a `VK_ERROR_DEVICE_LOST`, after about 12 minutes of execution:
```
...
[ RUN      ] VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB
C:\j\vz5\Release32\Vulkan-ValidationLayers\tests\vklayertests_gpu.cpp(1151): error: Expected equality of these values:
  VK_SUCCESS
    Which is: 0
  resolved_err
    Which is: -4
VK_ERROR_DEVICE_LOST
```
The test process continues to consume about 6% CPU, but making no progress, until it is killed.  This suggests that the test code itself is to blame for the hang (likely it doesn't expect or react to `VK_ERROR_DEVICE_LOST`).

## To Reproduce

Build **Vulkan-ValidationLayers** both in 32-bit and 64-bit; the instructions that follow assume the 32-bit build is in `build32\` and the 64-bit build is in `build\`.  Make sure the tests are built.

In one Command Prompt window, start the 64-bit tests.  The `--gtest_filter` switch is normally used to eliminate tests known to crash or fail; the list of tests below are the tests known to fail using this device and driver.  Unexpectedly, the `--gtest_filter` switch seems to be necessary to provoke the problem:
```
> cd ...\Vulkan-ValidationLayers
> set VK_LAYER_PATH=...\Vulkan-ValidationLayers\build\layers\Release
> build\tests\Release\vk_layer_validation_tests.exe --gtest_filter=-VkLayerTest.InvalidDescriptorSetLayoutBinding:VkLayerTest.CopyMutableDescriptors:VkLayerTest.ValidateUpdatingMutableDescriptors:VkLayerTest.MutableDescriptorPoolsWithPartialOverlap:VkLayerTest.InvalidWriteMutableDescriptorSet:VkLayerTest.MutableDescriptors:VkLayerTest.DescriptorUpdateTemplate:VkLayerTest.MutableDescriptorSetLayout:VkLayerTest.DescriptorBufferSetLayout:VkLayerTest.DescriptorBufferNotEnabled:VkLayerTest.DescriptorBufferDescriptorGetInfo:VkPositiveLayerTest.CopyMutableDescriptors
```
In another Command Prompt window, start the 32-bit tests with the same `--gtest_filter`:
```
> cd ...\Vulkan-ValidationLayers
> set VK_LAYER_PATH=...\Vulkan-ValidationLayers\build32\layers\Release
> build32\tests\Release\vk_layer_validation_tests.exe --gtest_filter=-VkLayerTest.InvalidDescriptorSetLayoutBinding:VkLayerTest.CopyMutableDescriptors:VkLayerTest.ValidateUpdatingMutableDescriptors:VkLayerTest.MutableDescriptorPoolsWithPartialOverlap:VkLayerTest.InvalidWriteMutableDescriptorSet:VkLayerTest.MutableDescriptors:VkLayerTest.DescriptorUpdateTemplate:VkLayerTest.MutableDescriptorSetLayout:VkLayerTest.DescriptorBufferSetLayout:VkLayerTest.DescriptorBufferNotEnabled:VkLayerTest.DescriptorBufferDescriptorGetInfo:VkPositiveLayerTest.CopyMutableDescriptors
```
After about 12 minutes, when the 32-bit tests attempt to run `VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB`, the test will show the `VK_ERROR_DEVICE_LOST` and then will hang.

### 32-bit `VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB`

I suspected there was something about the `VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB` itself, so in a second experiment I ran the 64-bit tests normally, but only ran the above test repeatedly on 32-bit:
```
> build32\tests\Release\vk_layer_validation_tests.exe --gtest_filter=VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB --gtest_repeat=10000
```
but the problem did not appear.  This suggests that there's a more complex interaction that involves the execution of other tests in addition to `VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB`.  I have not attempted to determine the exact set of tests involved.

### Eliminate the `--gtest_filter`

I hypothesized that the `--gtest_filter` switch was not necessary to observe the problem, and attempted to run both the 32-bit and 64-bit tests without it:
```
> build32\tests\Release\vk_layer_validation_tests.exe
```
```
> build\tests\Release\vk_layer_validation_tests.exe
```
Unexpectedly, without the `--gtest_filter` switch, the problem did not appear; both 32-bit and 64-bit tests successfully ran `VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB` and finished without problems.

This suggests that the problem is tied to some exact pattern of test execution.  It may suggest that the problem may not appear on slightly different builds, or that the problem might involve memory corruption or initialization done by earlier tests (though since the problem was never observed on driver 472.92, any such issue must be quite subtle).  Somehow the tests in the `gtest_filter` must also be involved.

**Comment by juan-lunarg** [comment]
Did #4969 help?

**Comment by lunarpapillo** [comment]
> Did #4969 help?

It did.  Testing with #4969 was clean - I was unable to reproduce the `VK_ERROR_DEVICE_LOST` issue.

I'm not hugely confident in this driver, which still seems quirky, but I'm willing to continue tests with it.  Closing.

**Comment by lunarpapillo** [comment]
FYI, the problem did appear later on driver 527.97 (despite #4969 ).  The failure appears to be sporadic, and still seems to be a driver issue.

I've reduced the executor count on `tcwinnvi2` from 6 to 1, which prevents both 64-bit and 32-bit tests from running at the same time; the loss in throughput is marginal since we've already serialized VVL jobs and performance is gated on Android runs, not Windows runs. 

We're also in discussion as to whether to blacklist `VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB` and allow concurrent runs, which may still cause sporadic failures to appear on this driver, or may allow tests to run concurrently without problem.

Reopening.
